name_similarity_parameters.rb

```diff
@@ -10,21 +10,24 @@ def initialize(name1, name2)
   end
 
   def validate_params
-    if [String, NameParameter].count { |clazz| @name1.instance_of? clazz } == 0
-      raise RosetteAPIError.new('badRequest', 'name1 option can only be an instance of a String or NameParameter')
-    elsif [String, Hash].count { |clazz| @name2.instance_of? clazz } == 0
-      raise RosetteAPIError.new('badRequest', 'name2 option can only be an instance of a String or NameParameter')
+    if [String, NameParameter].none? { |clazz| @name1.is_a? clazz }
```

Comment:
`#none?` returns true if none of the given blocks return `true`.

See http://devdocs.io/ruby~2.2/enumerable#method-i-none-3F for more details.